NOTES.WIN

```diff
@@ -93,9 +103,28 @@
    "vcvarsall.bat" before you compile. For example, if you want to build
    "arm64" builds, you should type "vcvarsall.bat x86_arm64 uwp".
 
- mingw and mingw64
+ MinGW and MinGW64
  =================
 
+ In the nutshell MinGW builds are always cross-compiled.
+ On Linux and Cygwin they look exactly as such and require
+ --cross-compile-prefix option. While on MSYS[2] it is solved
+ rather by placing gcc that produces "MinGW binary" code 1st on $PATH.
+ This is customarily source of confusion. "Hosted" applications "live" in
+ emulated file system name space with POSIX-y root, mount points, /dev
+ and even /proc. Confusion is intensified by the fact that MSYS2 shell
+ (or rather emulated execve(2) call) examines the binary it's about to
+ start, and if it's found *not* to be linked with MSYS2 POSIX-y thing,
+ command line arguments that look like file names get translated from
+ emulated name space to "native". For example '/c/some/where' becomes
+ 'c:\some\where', '/dev/null' - 'nul'. This creates an illusion that
+ there is no difference between MSYS2 shell and "MinGW binary", but
+ there is. Just keep in mind that "MinGW binary" "experiences" Windows
+ system in exactly same way as one produced by VC, and in its essence
+ is indistinguishable from the latter. (Which by the way is why
+ it's referred to in quotes here, as "MinGW binary", it's just as
+ "native" as it can get.)
```

Comment:
I think most of this is unnecessary. How about:

> MinGW offers an alternative way to build a native OpenSSL. 
> 
>  Requirement details
>  --------------------------
> 
> - MSYS2 shell. You must install the MSYS2 environment and start an MSYS2 shell. MSYS2 provides a Unix like command prompt, and a UNIX compatibility layer for applications. However in this context it is only used for building OpenSSL. The resulting OpenSSL that is created does not rely on MSYS2 to run and is fully native.
> 
> - MinGW or MinGW64 compiler. These compilers must be on your MSYS2 $PATH. A common error is to not have these on your $PATH and instead have the MSYS2 version of gcc. This will not work correctly.
> 
> - Perl, installed into the MSYS2 environment
>
> - make installed into the MSYS2 environment

Reply:
> I think most of this is unnecessary.

Yes, and also according to @levitte's comment I remove most of the MSYS details
and combined your suggesting with part of the remaining MinGW related text.